File: LeetCode-Algorithm/0026. Remove Duplicates from Sorted Array/Solution.java
package ja;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;


class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }

        int j = 0;
        for (int i = 1; i < nums.length; ++i) {
            if (nums[i] != nums[j]) {
                nums[++j] = nums[i];
            }
        }
        return j + 1;
    }
}

File: LeetCode-Algorithm/0026. Remove Duplicates from Sorted Array/solution.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        index = 0
        for x in nums:
            if x != nums[index]:
                index += 1
                nums[index] = x
        print(nums)
        return index + 1


if __name__ == '__main__':
    assert 2 == Solution().removeDuplicates([1, 1, 2])
    assert 5 == Solution().removeDuplicates([0, 0, 1, 1, 1, 2, 2, 3, 3, 4])
